libraries/Bluefruit52Lib/src/services/BLEHidAdafruit.cpp

```diff
@@ -51,24 +51,27 @@ uint8_t const hid_report_descriptor[] =
 {
   TUD_HID_REPORT_DESC_KEYBOARD( HID_REPORT_ID(REPORT_ID_KEYBOARD) ),
   TUD_HID_REPORT_DESC_CONSUMER( HID_REPORT_ID(REPORT_ID_CONSUMER_CONTROL) ),
-  TUD_HID_REPORT_DESC_MOUSE   ( HID_REPORT_ID(REPORT_ID_MOUSE) )
+  TUD_HID_REPORT_DESC_MOUSE   ( HID_REPORT_ID(REPORT_ID_MOUSE) ),
+  TUD_HID_REPORT_DESC_GAMEPAD ( HID_REPORT_ID(REPORT_ID_GAMEPAD) )
 };
 
 BLEHidAdafruit::BLEHidAdafruit(void)
-  : BLEHidGeneric(3, 1, 0)
+  : BLEHidGeneric(4, 1, 0)
 {
   _mse_buttons = 0;
+  _gpd_buttons = 0;
   _kbd_led_cb = NULL;
 }
 
 err_t BLEHidAdafruit::begin(void)
 {
-  uint16_t input_len [] = { sizeof(hid_keyboard_report_t),  sizeof(hid_consumer_control_report_t), sizeof(hid_mouse_report_t) };
+  uint16_t input_len [] = { sizeof(hid_keyboard_report_t),  sizeof(hid_consumer_control_report_t), sizeof(hid_mouse_report_t), sizeof(hid_gamepad_report_t) };
   uint16_t output_len[] = { 1 };
 
   setReportLen(input_len, output_len, NULL);
   enableKeyboard(true);
   enableMouse(true);
+  enableGamepad(true);
```

Comment:
Can you test to see if gamepad work with iOS. The reason I didn't add it in the first place is lacking support from iOS. Last time I tested (more than a year ago) iOS will disable device as a whole when it detect gamepad profile which cause keyboard and mouse not functioning at all. However, it is pre-iOS 13 and thing can be very different now. If you don't have any iOS device, no problem at all, I will do the test.

Reply:
I just can confirm you the whole code is fully working on Android, I have my custom gamepad fully working on Android 10. On an iOS 10 iPad my custom controller is paired successfully but it doesn't have any effect by using the gamepad report. My PS4 controller is not even recognized on the iOS10 bluetooth menu, I can't pair it. I heard something about controllers on iOS must be MFI (Made For iOS), maybe they have some kind of non standard special BLE protocol, I don't know nothing about it. Although we can't make this work with iOS it fully works with Windows, Linux and Android, three different OS, and the custom BLE gamepad report is more important to achive for any RasberryPi rather than any iOS device.